CommandLine/libEGM/gmx.cpp

```diff
@@ -255,7 +255,7 @@ void PackRes(std::string &dir, int id, pugi::xml_node &node, google::protobuf::M
     const google::protobuf::FieldOptions opts = field->options();
 
     if (field->name() == "id") {
-      id += opts.GetExtension(buffers::id_start);
+      int id = opts.GetExtension(buffers::id_start) + ids[m->GetTypeName()]++;
```

Comment:
This is slow; please cache this by reference at the top of the method

Reply:
Which part of it? It may be difficult as both Tile and Instance message use this same option within the same proto.

Reply:
The lookup of `ids[m->GetTypeName()]`. You can stash that as a const-ref.

Reply:
Stash it where?

Reply:
> at the top of the method

Reply:
@fundies you didn't fix this before merging!